Drivers/Src/Display.c

```diff
@@ -1,325 +1,107 @@
-        return setComponentVisibility(REGEN_ENABLE, true);
-    }
-    else
-    {
-        return setComponentVisibility(REGEN_ENABLE, false);
-    }
-}
-
-/**
- * Set the displayed supplemental battery pack voltage.
- * Units of millivolts
- */
-ErrorStatus Display_SetSBPV(uint16_t mv)
-{
-    int32_t sbpv = mv / 100; // One tenth of a volt precision
-    return updateIntValue(SUPPL_VOLT, VALUE, sbpv);
-}
-
-/**
- * Update the cruise enable light on the display based on the given state
- */
-ErrorStatus Display_CruiseEnable(State on)
-{
-    if (on == ON)
-    {
-        return setComponentVisibility(CRUISE_ENABLE, true);
-    }
-    else
-    {
-        return setComponentVisibility(CRUISE_ENABLE, false);
-    }
-}
-
-ErrorStatus Display_SetGear(State fwd, State rev){
-    if(!fwd && !rev){
-        setComponentVisibility(NEUTRAL_BOX, true);
-        setComponentVisibility(FWD_BOX, false);
-        return setComponentVisibility(REV_BOX, false);
-    }
-    else if(fwd && !rev){
-        setComponentVisibility(NEUTRAL_BOX, false);
-        setComponentVisibility(FWD_BOX, true);
-        return setComponentVisibility(REV_BOX, false);
-    }
-    else if(rev && !fwd){
-        setComponentVisibility(NEUTRAL_BOX, false);
-        setComponentVisibility(REV_BOX, true);
-        return setComponentVisibility(FWD_BOX, false);
-    }
-    else{
-        return ERROR;
-    }
-}
-
-
-/**
- * Set ERROR<idx> to err
- * If err is an empty string, then the error will be cleared
- */
-ErrorStatus Display_SetError(int idx, char *err)
-{
-    if (idx < 0 || idx > 5)
-        return ERROR; // Index out of bounds
-    if (strlen(err) == 0)
-    {
-        CHECK(setComponentVisibility(ERROR0 + idx, false)); // Hide text if no error
-    }
-    else
-    {
-        CHECK(updateStringValue(ERROR0 + idx, TEXT, err)); // Set error string
-        CHECK(updateIntValue(ERROR0 + idx, PCO, NEXTION_RED));
-        CHECK(setComponentVisibility(ERROR0 + idx, true));
-    }
-    return SUCCESS;
-}
-
-/**
- * Set the first error slot to say "No Error" and set its color to green
- * User must clear the remaining slots manually using Display_SetError
- */
-ErrorStatus Display_NoErrors(void)
-{
-    CHECK(updateStringValue(ERROR0, TEXT, (char *)NO_ERROR));
-    CHECK(updateIntValue(ERROR0, PCO, NEXTION_GREEN));
-    CHECK(setComponentVisibility(ERROR0, true));
-    return SUCCESS;
-}
-
-/**
- * Set the display to the main view
- */
-ErrorStatus Display_SetMainView(void)
-{
-    //return updateIntValue(SYSTEM, PAGE, 1);
-    char *page = "page1\xff\xff\xff";
-    BSP_UART_Write(DISP_OUT, page, strlen(page));
-    return SUCCESS;
-}
-
-ErrorStatus Display_SetLight(uint8_t light, State on){
-    if(light == 2){  // ctrl fault
-        if(on==ON)
-            return updateIntValue(PAGE, BCO, NEXTION_RED);
-        return updateIntValue(PAGE, BCO, NEXTION_DARK_GREY);
-    }
-        
-    setComponentVisibility(light+19, on==ON);
-    return SUCCESS;
-}
+ErrorStatus Display_Init(){
```

Comment:
Sending a display reset command might be good here, since resetting the Leaderboard doesn't reset the display as well.